docs/book/getting-started/database-and-models.md

```diff
@@ -199,74 +199,111 @@ of these methods is, hopefully, self-explanatory.
 ## Using ServiceManager to configure the table gateway and inject into the AlbumTable
 
 In order to always use the same instance of our `AlbumTable`, we will use the
-`ServiceManager` to define how to create one. This is most easily done in the
-`Module` class where we create a method called `getServiceConfig()` which is
-automatically called by the `ModuleManager` and applied to the `ServiceManager`.
-We'll then be able to retrieve when we need it.
+`ServiceManager` to define how to create one. 
+This is most easily done by adding a `ServiceManager` configuration to the `module.config.php` 
+which is automatically loaded by the `ModuleManager` and applied to the `ServiceManager`.
+We'll then be able to retrieve the `AlbumTable` when we need it.
 
 To configure the `ServiceManager`, we can either supply the name of the class to
-be instantiated or a factory (closure, callback, or class name of a factory
-class) that instantiates the object when the `ServiceManager` needs it. We start
-by implementing `getServiceConfig()` to provide a factory that creates an
-`AlbumTable`. Add this method to the bottom of the `module/Album/src/Module.php`
-file:
+be instantiated and a factory (closure, callback, or class name of a factory
+class) that instantiates the object when the `ServiceManager` needs it. 
+
+Add a `service_manager` configuration to `module/Album/config/module.config.php`:
 
 <!-- markdownlint-disable MD033 -->
-<pre class="language-php" data-line="3-6,13-30"><code>
+<pre class="language-php" data-line="3,38-41"><code>
 namespace Album;
 
-// Add these import statements:
+use Album\Model\AlbumTableFactory;
+use Laminas\Router\Http\Segment;
+use Laminas\ServiceManager\Factory\InvokableFactory;
+
+return [
+    'controllers' => [
+        'factories' => [
+            Controller\AlbumController::class => InvokableFactory::class,
+        ],
+    ],
+
+    // The following section is new and should be added to your file:
+    'router' => [
+        'routes' => [
+            'album' => [
+                'type'    => Segment::class,
+                'options' => [
+                    'route' => '/album[/:action[/:id]]',
+                    'constraints' => [
+                        'action' => '[a-zA-Z][a-zA-Z0-9_-]*',
+                        'id'     => '[0-9]+',
+                    ],
+                    'defaults' => [
+                        'controller' => Controller\AlbumController::class,
+                        'action'     => 'index',
+                    ],
+                ],
+            ],
+        ],
+    ],
+    'view_manager' => [
+        'template_path_stack' => [
+            'album' => __DIR__ . '/../view',
+        ],
+    ],
```

Comment:
This part was already added in: https://docs.laminas.dev/tutorials/getting-started/routing-and-controllers/#routing-and-controllers

So this can be reduced to:

```suggestion
    'controllers' => [
        // …
    ],
    'router' => [
        // …
    ],
    'view_manager' => [
        // …
    ],
```

Reply:
Done